projects/online/online/dataloading/offline.py

```diff
@@ -0,0 +1,302 @@
+import os
+import numpy as np
+from pathlib import Path
+from online.dataloading.utils import (
+    build_resample_filter,
+    resample,
+    parse_frame_name,
+    GWF_SAMPLE_RATE,
+)
+from typing import List, Optional, Generator
+from gwpy.timeseries import TimeSeriesDict
+import logging
+import torch
+import multiprocessing as mp
+
+
+STATE_VECTOR_SAMPLE_RATE = 16
+
+
+class OfflineFrameFileLoader:
+    def __init__(
+        self,
+        datadir: Path,
+        ifo_suffix: str,
+        ifos: list[str],
+        strain_channels: list[str],
+        state_channels: list[str],
+    ):
+        """
+        Initialize the chunk generator for timeseries data.
+
+        Args:
+            directories: List of directory paths containing timeseries files
+            sample_rate: Sample rate of the data in Hz
+        """
+        self.strain_channels = strain_channels
+        self.state_channels = state_channels
+        self.directories: dict[str, Path] = {
+            ifo: datadir / f"{ifo}_{ifo_suffix}" for ifo in ifos
+        }
+        self.ifos = ifos
+        self.frame_metadata = {}
+        self.start, self.end = self.parse_directories()
+
+    def parse_directories(self):
+        """Scan directories and organize file information by timestamp."""
+        for ifo, path in self.directories.items():
+            self.frame_metadata[ifo] = {}
+
+            for filename in path.iterdir():
+                if filename.suffix == ".gwf":
+                    _, timestamp, duration = parse_frame_name(filename)
+                    self.frame_metadata[ifo][timestamp] = {
+                        "path": os.path.join(path, filename),
+                        "end": timestamp + duration,
+                    }
+
+        # Find the common time range across all directories
+        return self.find_intersection()
```

Comment:
Is this necessary, or just a precaution? I would have expected the MDC frame directories to contain all the same frame files.

Reply:
This was more of a precaution.